db/db_client/db_client_session.go

```diff
@@ -8,57 +8,84 @@ import (
 	"time"
 
 	"github.com/sethvargo/go-retry"
+	"github.com/turbot/steampipe/db/db_common"
 	"github.com/turbot/steampipe/utils"
 )
 
-func (c *DbClient) AcquireSession(ctx context.Context) (_ *sql.Conn, acquireSessionError error) {
+func (c *DbClient) AcquireSession(ctx context.Context) (_ *db_common.DBSession, acquireSessionError error) {
 	c.sessionInitWaitGroup.Add(1)
-	c.sessionAcquireLock.Lock()
-
-	defer func() {
-		c.sessionInitWaitGroup.Done()
-		c.sessionAcquireLock.Unlock()
-	}()
+	defer c.sessionInitWaitGroup.Done()
 
 	// get a database connection and query its backend pid
 	// note - this will retry if the connection is bad
-	session, backendPid, err := c.getSessionWithRetries(ctx)
+	rawConnection, backendPid, err := c.getSessionWithRetries(ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	c.sessionsMutex.Lock()
+	session, found := c.sessions[backendPid]
+	if !found {
+		session = db_common.NewDBSession(backendPid)
+	}
+	// we get a new *sql.Conn everytime. USE IT!
+	session.Raw = rawConnection
```

Comment:
maybe call the property `connection` - I agree this is overloaded but in the context of the session it is fairly clear
`session.Connection = databaseConnection`